packages/react-query/src/__tests__/queryOptions.test-d.tsx

```diff
@@ -236,7 +237,9 @@ describe('queryOptions', () => {
   })
 
   it('should be passable to UseQueryOptions', () => {
-    function somethingWithQueryOptions(options: UseQueryOptions<number>) {
+    function somethingWithQueryOptions<TQueryOpts extends AnyUseQueryOptions>(
+      options: TQueryOpts,
+    ) {
```

Comment:
Hey @TkDodo so I think the other two approaches proved to be dead ends, and I'm also unsure about making major internal changes like those as they risk creating a new wave of breaking issues.

What I propose is we export types like `AnyUseQueryOptions`, and that way consumers wanting to create flexible re-usable methods can use a more correct form of generics like above (don't necessarily need the generics but for anything non-contrived it's probably useful). This is how tRPC and a few other projects solve this issue and it's rock solid